File: 250515/특정 구간의 원소 평균값/elemental-mean-value-for-a-particular-interval.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++)
            arr[i] = sc.nextInt();
        // Please write your code here.

        int count = 0;
        for (int i = 0; i < n; i++) {
            for (int j = i; j < n; j++) {
                double avg = 0;
                for (int k = i; k <= j; k++) {
                    avg += arr[k];
                }
                avg = avg / (double) (j-i+1);

                for (int k = i; k <= j; k++) {
                    if ((double) arr[k] == avg) {
                        count++;
                        break;
                    }
                }
            }
        }
        System.out.print(count);
    }
}

// 

File: 251024/특정 구간의 원소 평균값/elemental-mean-value-for-a-particular-interval.py
n = int(input())
arr = list(map(int, input().split()))

# Please write your code here.

# 평균값 = 그 구간의 원소 중 하나
# 서로 다른 가짓수
resultCnt = 0
for i in range(n) :
    for j in range(i,n):
        
        sum = 0 
        cnt = 0

        for k in range(i,j+1) :
            sum += arr[k] 
            cnt += 1

        for k in range(i, j+1) :
            if arr[k] == sum/cnt :
                resultCnt += 1
                break

print(resultCnt)
        